disco/extensions/upgrade_simulation/upgrades/voltage_upgrade_functions.py

```diff
@@ -885,36 +924,111 @@ def determine_new_regulator_upgrades(voltage_config, buses_with_violations, volt
     if comparison_dict["after_addition_new_regcontrol"][deciding_field] < comparison_dict[best_setting_so_far][deciding_field]:
         best_setting_so_far = "after_addition_new_regcontrol"
+        new_reg_upgrade_commands = reg_upgrade_commands
     else: 
         new_reg_upgrade_commands = []
-        dss_commands_list = list(set(dss_commands_list) - set(regcontrol_settings_commands_list) - set(regcontrol_cluster_commands))
+        remove_commands_list = reg_upgrade_commands
+        # TODO: better to build a temporary list of the names or indices to remove and then iterate over that to perform the removals.
+        if remove_commands_list:
+            for x in dss_commands_list:
+                if x in remove_commands_list:
+                    dss_commands_list.remove(x)
         reload_dss_circuit(dss_file_list=dss_file_list, commands_list=dss_commands_list, **kwargs)
         comparison_dict["disabled_new_regcontrol"] = compute_voltage_violation_severity(
                                                         voltage_upper_limit=voltage_upper_limit, voltage_lower_limit=voltage_lower_limit, **kwargs)
 
     return {"new_reg_upgrade_commands": new_reg_upgrade_commands, "comparison_dict": comparison_dict, "best_setting_so_far": best_setting_so_far}
 
 
+def get_newly_added_regulator_settings(dss_file_list, previous_dss_commands_list, regcontrol_cluster_commands, voltage_lower_limit, voltage_upper_limit,
+                                       voltage_config, deciding_field, **kwargs):
+    """this function finalizes the settings with the newly added voltage regulator. 
+    It also takes into account errors encountered due to max control iterations being exceeded.
+    """
+    # reload circuit after settings sweep
+    try:
+        logger.info("Settings sweep for existing reg control devices (other than sub LTC).")
+        regcontrol_df = get_regcontrol_info()
+        regcontrol_sweep_df = sweep_regcontrol_settings(voltage_config=voltage_config,
+                                                        initial_regcontrols_df=regcontrol_df,
+                                                        voltage_upper_limit=voltage_upper_limit, voltage_lower_limit=voltage_lower_limit,
+                                                        exclude_sub_ltc=True, only_sub_ltc=False, **kwargs)
+        regcontrols_df, regcontrol_settings_commands_list = choose_best_regcontrol_sweep_setting(deciding_field=deciding_field,
+            regcontrol_sweep_df=regcontrol_sweep_df, initial_regcontrols_df=regcontrol_df, **kwargs)
+        reload_dss_circuit(dss_file_list=dss_file_list, commands_list=previous_dss_commands_list + regcontrol_cluster_commands + regcontrol_settings_commands_list, **kwargs)  # reload circuit after settings sweep
+        reg_upgrade_commands = regcontrol_cluster_commands + regcontrol_settings_commands_list
+    except Exception as e:  # if there is an error: dss._cffi_api_util.DSSException: (#485)

```

Comment:
This except is too broad. You have quite a bit of code in the try block (and in the called functions). If you have a syntax error, it will fall into the except block. Are you only trying to catch errors in `reload_circuit`? You could only wrap that function. Even better, only catch the exceptions like the one you list.

Reply:
Added better exception handling